napari/plugins/__init__.py

```diff
@@ -26,6 +29,28 @@
     plugin_manager.register(_builtins, name='builtins')
 
 
+dock_widgets: Dict[str, Type[QWidget]] = dict()
+
+
+def register_dock_widget(cls: Union[Type[QWidget], Sequence[Type[QWidget]]]):
+    for _cls in cls if is_sequence(cls) else [cls]:
+        if not isclass(_cls) and issubclass(_cls, QWidget):
+            # what to do here?
+            continue
+        name = getattr(
+            _cls, 'napari_menu_name', camel_to_spaces(_cls.__name__)
+        )
+        if name in dock_widgets:
+            # duplicate menu names... what to do here?
+            continue
+        dock_widgets[name] = _cls
+
+
+plugin_manager.hook.napari_experimental_provide_dock_widget.call_historic(
+    result_callback=register_dock_widget
+)
```

Comment:
@tlambert03 I'm working on a follow-up PR to this adding in the function hook spec, and I'd like to be able to pass the plugin name to `register_dock_widget` too, but can't figure out how to do it! Any ideas? This would let us "name space" the keys above